app/src/main/java/uw/hcrlab/kubi/lesson/PromptData.java

```diff
@@ -47,27 +47,76 @@ public String toString() {
                 this.type, this.PromptText, wordString, optionsString, imagesString);
     }
 
+    public static class Hint {
+        // will extend so this can be an "explain" or "conjugate" button when applicable
+        public String text;
+
+        public Hint setText(String text) {
+            this.text = text;
+            return this;
+        }
+
+        public String toString() {
+            return String.format(Locale.US, "Hint {%s}", this.text);
+        }
+    }
+
+    public static class HintCollection {
+        public ArrayList<Hint> details;
+
+        public HintCollection() {
+            this.details = new ArrayList<>();
+        }
+
+        public String toString() {
+            String detailString = "";
+            for (Hint detail: this.details) {
+                detailString += detail.toString() + ", ";
+            }
+            return String.format(Locale.US, "HintCollection {details=%s}", detailString);
+        }
+
+        public boolean isEmpty() {
+            return this.details.size() < 1;
+        }
+    }
+
```

Comment:
I guess I'd defer to you on this since I come from python where everyone has access to everything :)

But I think if any class has first priority for getting split up into multiple classes, it should be Robot.